### PR TITLE
Add a command to set DirectChildrenLoaded on dir

### PR DESCRIPTION
**Pull request by maobaolong**
```
$ bin/alluxio fs syncDirNextTime true /data
Successfully marked the dir /data to trigger metadata sync on next access

$ bin/alluxio fs syncDirNextTime false /data
Successfully marked the dir /data to skip metadata sync on next access
```

### CODE REVIEW

**Comment by dbw9580** [comment]
context is in #15647

**Comment by dbw9580** [comment]
I didn't quite follow the original issue in #15647, but if the decision is that a user facing command is desired, then we need to come up with a better name than `directChildrenLoaded` and clear definition of what this does.

File: shell/src/main/java/alluxio/cli/fs/command/SetDirectChildrenLoadedCommand.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import java.io.IOException;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Sets direct children loaded command.
+ */
+@ThreadSafe
+@PublicApi
+public final class SetDirectChildrenLoadedCommand extends AbstractFileSystemCommand {
```
[the remark this answers is not in the document]
Comment (jiacheliu3) [reply shell/src/main/java/alluxio/cli/fs/command/SetDirectChildrenLoadedCommand.java:31]: or `SetReloadCommand`

**Comment by maobaolong** [comment]
@jiacheliu3 @dbw9580 @elega  Thanks for the discussion about this PR, I rename the command, PTAL

File: shell/src/main/java/alluxio/cli/fs/command/SyncNextTimeCommand.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  @Override
+  public void validateArgs(CommandLine cl) throws InvalidArgumentException {
+    CommandUtils.checkNumOfArgsEquals(this, cl, 2);
+  }
+
+  @Override
+  protected void runPlainPath(AlluxioURI path, CommandLine cl)
+      throws AlluxioException, IOException {
+    FileSystemCommandUtils.setDirectChildrenLoaded(mFileSystem, path, mLoaded);
+    System.out.println("Path '" + path + "' was successfully set sync direct children to "
```
Comment (jiacheliu3) [inline shell/src/main/java/alluxio/cli/fs/command/SyncNextTimeCommand.java:56]: based on setting to `true` or `false`, can you pls print a more meaningful message?
True:  `syncNextTime path true` means the path will be sync-ed next time so you print
`Successfully marked the dir {} to trigger metadata sync on next access`

False: `syncNextTime path false` means the next sync is disabled so you print
`Successfully marked the dir {} to skip metadata sync on next access`

File: shell/src/main/java/alluxio/cli/fs/command/SyncNextTimeCommand.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  @Override
+  public String getUsage() {
+    return "syncNextTime <true|false> <path>";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Sets sync direct children next time for a specific directory to true or false";
```
Comment (jiacheliu3) [inline shell/src/main/java/alluxio/cli/fs/command/SyncNextTimeCommand.java:75]: "Marks a directory to either trigger a metadata sync or skip the metadata sync on next access."

File: shell/src/main/java/alluxio/cli/fs/command/SyncNextTimeCommand.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+import java.io.IOException;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Sync direct children next time command.
+ */
+@ThreadSafe
+@PublicApi
+public final class SyncNextTimeCommand extends AbstractFileSystemCommand {
```
Comment (jiacheliu3) [inline shell/src/main/java/alluxio/cli/fs/command/SyncNextTimeCommand.java:31]: Actually this only works on directory, so `SyncDirNextTimeCommand` and `syncDirNextTime`?

**Review by jiacheliu3 — COMMENTED** [review]
Mostly LGTM with some minor comments on naming and outputs. Also is it possible to add some UTs? Thanks!

File: shell/src/main/java/alluxio/cli/fs/command/SyncDirNextTimeCommand.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  public int run(CommandLine cl) throws AlluxioException, IOException {
+    String[] args = cl.getArgs();
+    mLoaded = Boolean.parseBoolean(args[0]);
+    runWildCardCmd(new AlluxioURI(args[1]), cl);
+    return 0;
+  }
+
+  @Override
+  public String getUsage() {
+    return "syncNextTime <true|false> <path>";
```
Comment (jiacheliu3) [inline shell/src/main/java/alluxio/cli/fs/command/SyncDirNextTimeCommand.java:71]:
```suggestion
    return "syncDirNextTime <true|false> <path>";
```

Comment (jiacheliu3) [reply shell/src/main/java/alluxio/cli/fs/command/SyncDirNextTimeCommand.java:71]: Also the `true|false` here is not obvious. Let's add
```
return "syncDirNextTime <true|false> <path>\n"
    + "\ttrue means the next access will trigger a metadata sync on the dir"
    + "\tfalse means the next metadata sync is disabled"
```

File: shell/src/main/java/alluxio/cli/fs/command/SyncDirNextTimeCommand.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    CommandUtils.checkNumOfArgsEquals(this, cl, 2);
+  }
+
+  @Override
+  protected void runPlainPath(AlluxioURI path, CommandLine cl)
+      throws AlluxioException, IOException {
+    FileSystemCommandUtils.setDirectChildrenLoaded(mFileSystem, path, mLoaded);
+    System.out.println("Successfully marked the dir {} to "
+        + (mLoaded ? "trigger metadata sync on next access"
+        : "skip metadata sync on next access"));
```
Comment (jiacheliu3) [inline shell/src/main/java/alluxio/cli/fs/command/SyncDirNextTimeCommand.java:58]:
```suggestion
    System.out.format("Successfully marked the dir %s to %s%n", path, 
        mLoaded ? "trigger metadata sync on next access" : "skip metadata sync on next access"));
```

File: shell/src/main/java/alluxio/cli/fs/command/SyncDirNextTimeCommand.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Sync direct children next time command.
+ */
+@ThreadSafe
+@PublicApi
+public final class SyncDirNextTimeCommand extends AbstractFileSystemCommand {
+
+  private boolean mLoaded;
```
Comment (jiacheliu3) [inline shell/src/main/java/alluxio/cli/fs/command/SyncDirNextTimeCommand.java:33]:
```suggestion
  private boolean mSyncNextTime;
```

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Mostly LGTM with minor comments. Thanks for the work!

File: shell/src/main/java/alluxio/cli/fs/command/SyncDirNextTimeCommand.java (hunk 6 above)
Comment (maobaolong) [reply shell/src/main/java/alluxio/cli/fs/command/SyncDirNextTimeCommand.java:58]: done

File: shell/src/main/java/alluxio/cli/fs/command/SyncDirNextTimeCommand.java (hunk 5 above)
Comment (maobaolong) [reply shell/src/main/java/alluxio/cli/fs/command/SyncDirNextTimeCommand.java:71]: done

**Comment by maobaolong** [comment]
@jiacheliu3 Thanks for the kindly review, I address all your suggestion, PTAL.

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks for the contribution!

Ideally we have the doc change and associated UTs, but that can be done in separate PRs.

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**